DVSRegister/Validations/ContactViewModelValidator.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Text.RegularExpressions;
+using PhoneNumbers;
+
+namespace DVSRegister.Validations
+{
+	public class ContactViewModelValidator
```

Comment:
can we modify the validator to update context directly?   ValidateSponsorViewModel and ValidateContactViewModel can be called from controller. all existing methods in the validator can be changed to private
 public class ContactViewModelValidator
 {

     //Validate Sponsor model call method to validate contact model
     public static void ValidateSponsorViewModel(HttpContext httpContext, SponsorViewModel sponsorViewModel)
     {
         httpContext?.Session.Set("IsSponsorFullNameValid", ValidateFullName(sponsorViewModel?.SponsorFullName));
         httpContext?.Session.Set("IsSponsorEmailValid", ValidateEmail(sponsorViewModel?.SponsorEmail));
        // add reemaining fields
         //call method to validate contact model
         ValidateContactViewModel(httpContext, sponsorViewModel.ContactViewModel);

     }

     /// <summary>
     /// Validates the contact view model only. 
     /// </summary>
     /// <param name="httpContext">The HTTP context.</param>
     /// <param name="contactViewModel">The contact view model.</param>
     public static void ValidateContactViewModel(HttpContext httpContext, ContactViewModel contactViewModel)
      {
             httpContext?.Session.Set("IsFullNameValid", ValidateFullName(contactViewModel.FullName));
             httpContext?.Session.Set("IsEmailValid", ValidateEmail(contactViewModel.Email));
             // add other fields
      }

     /// <summary>
     /// Private method- Validates the full name.
     /// </summary>
     /// <param name="fullName">The full name.</param>
     /// <returns></returns>
     private static bool ValidateFullName(string fullName)
     {
         if (fullName != null && fullName.Length > 0)
         {
             return true;
         }
         else
         {
             return false;
         }
     }
}

Reply:
Done